core/src/test/kotlin/ch/ergon/dope/resolvable/expression/CaseExpressionTest.kt

```diff
@@ -0,0 +1,442 @@
+
+        assertEquals(expected.toDopeQuery(), actual.toDopeQuery())
+    }
+
+    @Test
+    fun `should support unaliased case expression function with expression`() {
+        val whenThenCondition = whenThen(someBooleanField(), someNumberField())
+        val expression = someStringField()
+        val expected = UnaliasedCaseExpression(
+            expression = expression,
+            whenThenCondition = whenThenCondition,
+        )
+
+        val actual = case(
+            expression = expression,
+            whenThenCondition = whenThenCondition,
+        )
+
+        assertEquals(expected.toDopeQuery(), actual.toDopeQuery())
+    }
+
+    @Test
+    fun `should support unaliased case expression function with expression and with else`() {
+        val whenThenCondition = whenThen(someBooleanField(), someNumberField())
+        val expression = someStringField()
+        val elseCase = someStringField()
+        val expected = UnaliasedCaseExpression(
+            expression = expression,
+            whenThenCondition = whenThenCondition,
+            elseCase = elseCase,
+        )
+
+        val actual = case(
+            expression = expression,
+            whenThenCondition = whenThenCondition,
+            elseCase = elseCase,
+        )
+
+        assertEquals(expected.toDopeQuery(), actual.toDopeQuery())
+    }
+
+    @Test
+    fun `should support unaliased case expression function and multiple when then`() {
+        val whenThenCondition = whenThen(someBooleanField(), someNumberField())
+        val additionalWhenThenConditions = listOf(whenThenCondition).toTypedArray()
+        val expected = UnaliasedCaseExpression(
+            whenThenCondition = whenThenCondition,
+            additionalWhenThenConditions = additionalWhenThenConditions,
+        )
+
+        val actual = case(
+            whenThenCondition = whenThenCondition,
+            additionalWhenThenConditions = additionalWhenThenConditions,
+        )
+
+        assertEquals(expected.toDopeQuery(), actual.toDopeQuery())
+    }
+
+    @Test
+    fun `should support unaliased case expression function with expression and multiple when then`() {
+        val whenThenCondition = whenThen(someBooleanField(), someNumberField())
+        val expression = someStringField()
+        val additionalWhenThenConditions = listOf(whenThenCondition).toTypedArray()
+        val expected = UnaliasedCaseExpression(
+            expression = expression,
+            whenThenCondition = whenThenCondition,
+            additionalWhenThenConditions = additionalWhenThenConditions,
+        )
+
+        val actual = case(
+            expression = expression,
+            whenThenCondition = whenThenCondition,
+            additionalWhenThenConditions = additionalWhenThenConditions,
+        )
+
+        assertEquals(expected.toDopeQuery(), actual.toDopeQuery())
+    }
+
+    @Test
+    fun `should support unaliased case expression function and multiple when then and else`() {
```

Comment:
The test does not test for multiple when then conditions

Reply:
done